docs/types.rst

```diff
@@ -0,0 +1,318 @@
+  * ``security_groups``: list of security group names.
+  * ``security_group_ids``: a list of security group IDs.
+
+If you want to specify the ``availability_zone`` for your instance, you must use the ``placement`` key.
+
+The create function also sets ``reservation_id`` attribute. For information, see [here](http://boto.readthedocs.org/en/latest/ref/ec2.html#boto.ec2.instance.Reservation)
+
+Four additional ``runtime_properties`` are available on node instances of this type once the ``cloudify.interfaces.lifecycle.start`` operation succeeds:
+
+  * ``ip`` the instance's private IP.
+  * ``private_dns_name`` the instance's private FQDN in Amazon.
+  * ``public_dns_name`` the instances's public FQDN in Amazon.
+  * ``public_ip_address`` the instance's public IP address.
+
+.. tip::
+    If you want to use the instance in VPC, then you need to connect this to a Subnet using the `cloudify.aws.relationships.instance_contained_in_subnet` relationship.
+
+.. cfy:node:: cloudify.aws.nodes.WindowsInstance
+
+    Use this type when working with a Windows server.
+    It has the same properties and operations-mapping as
+    :cfy:node:`cloudify.aws.nodes.Instance`,
+    yet it overrides some of the agent and plugin installations operations-mapping derived from
+    the :cfy:node:`cloudify.nodes.Compute` type.
+
+    Additionally, the default value for the ``use_password`` property
+    is overridden for this type, and is set to ``true``.
+    In this case, the password of the windows server will be retrieved,
+    decrypted and put under the ``password`` runtime property of this node instance.
+
+
+.. cfy:node:: cloudify.aws.nodes.KeyPair
+
+
+.. cfy:node:: cloudify.aws.nodes.SecurityGroup
+
+.. note::
+    If you want to create a security group in a VPC,
+    you need to connect it to a VPC using the
+    :cfy:rel:`cloudify.aws.relationships.security_group_contained_in_vpc`
+    relationship.
+
+
+.. cfy:node:: cloudify.aws.nodes.ElasticIP
+
+.. note::
+    the actual IP is available via the ``aws_resource_id`` runtime-property.
+
+
+.. cfy:node:: cloudify.aws.nodes.VPC
+
+    For more info on VPC, see https://aws.amazon.com/documentation/vpc/
+
+.. note::
+    When a VPC is created, it receives several default attachments.
+    We assign a runtime property for original dhcp options set,
+    called ``default_dhcp_options_id``.
+    Note that this is not necessarily the current dhcp options set.
+
+
+.. cfy:node:: cloudify.aws.nodes.Subnet
+
+
+.. cfy:node:: cloudify.aws.nodes.InternetGateway
+
+
+.. cfy:node:: cloudify.aws.nodes.VPNGateway
+
+
+.. cfy:node:: cloudify.aws.nodes.CustomerGateway
+
+
+.. cfy:node:: cloudify.aws.nodes.ACL
+
+
+.. cfy:node:: cloudify.aws.nodes.DHCPOptions
+
+
+.. cfy:node:: cloudify.aws.nodes.RouteTable
+.. cfy:node:: cloudify.aws.nodes.SecurityGroupRule
```

Comment:
I found a good number of node_types and relationships were not previously documented.
Please let me know if that was intentional. If so I will need to create some capability for `sphinxify` to mark types as hidden.
(Perhaps what we really need is a way to mark types as deprecated within the DSL?)